src/randoop/JunitFileWriter.java

```diff
@@ -195,58 +183,86 @@ public File writeDriverFile(String driverClassName) {
   }
   
-  public static File writeDriverFile(File dir, String packageName, String driverClassName,
-      List<String> junitTestSuiteNames) {
-    File file = new File(dir, driverClassName + ".java");
+  /*
+   * writes JUnit4 suite based on list of class names
+   */
+  public File writeSuiteFile (List<String> testClassNames) {
+    File dir = this.getDir();
+    String suiteClassName = masterTestClassName;
+    File file = new File(dir, suiteClassName+".java");
     PrintStream out = createTextOutputStream(file);
     try {
-      outputPackageName(out, packageName);
-      out.println("import junit.framework.*;");
-      out.println("import junit.textui.*;");
-      out.println("");
-      out.println("public class " + driverClassName + " extends TestCase {");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static void main(String[] args) {");
-      } else {
-      out.println("  public static void main(String[] args) throws Throwable {");
+      outputPackageName(out,packageName);
+      
+      out.println();
+      out.println("import org.junit.runner.RunWith;");
+      out.println("import org.junit.runners.Suite;");
+      out.println();
+      out.println("@RunWith(Suite.class)");
+      out.println("@Suite.SuiteClasses({");
+     
+      Iterator<String> testIterator = testClassNames.iterator();
+      if (testIterator.hasNext()) {
+        out.println(testIterator.next() + ".class");
+        while (testIterator.hasNext()) {
+          out.print(", ");
+          out.println(testIterator.next() + ".class");
+        }
       }
-      if (GenInputsAbstract.init_routine != null)
-        out.println ("    " + GenInputsAbstract.init_routine + "();");
-
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("    TestRunner runner = new TestRunner();");
-      out.println("    TestResult result = runner.doRun(suite(), false);");
-      out.println("    if (! result.wasSuccessful()) {");
-      out.println("      System.exit(1);");
-      out.println("    }");
-      } else {
-      out.println("    new " + driverClassName + "(\"" + driverClassName + "\").runTest();");
+      
+      out.println("})");
+      out.println("public class " + suiteClassName + "{ }");
+    } finally {
+      if (out != null) {
+        out.close();
       }
-      out.println("  }");
-      out.println("");
-      out.println("  public " + driverClassName + "(String name) {");
-      out.println("    super(name);");
-      out.println("  }");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static Test suite() {");
-      out.println("    TestSuite result = new TestSuite();");
-      for (String junitTestsClassName : junitTestSuiteNames) {
-        out.println("    result.addTest(new TestSuite(" + junitTestsClassName + ".class));");
+    }
+    return file;
+  }
+  
+  /*
+   * write non-reflective driver for tests
```

Comment:
Use capitalization and punctuation, here and elswhere, or the Javadoc will be unnecessarily hard to read.  Also use /** or else the comments won't be Javadoc comments at all.
You can use the Javadoc tag @see to refer to writeSubSuite.